util/iscn.js

```diff
@@ -1,220 +1,118 @@
+    iscnSigningClient = await getISCNSigningClient();
+    signingStargateClient = iscnSigningClient.getSigningStargateClient();
   }
-  return signingClient;
+  return signingStargateClient;
 }
 
-async function signISCNTx(inputPayload, explicitSignerData = null) {
-  const { wallet, account } = await getWallet();
-  const { iscnId, ...payload } = inputPayload;
-  const isUpdate = !!iscnId;
-  const record = formatISCNPayload(payload);
-  const client = await getSigningClient(wallet);
-
-  const value = {
-    from: account.address,
-    record,
-  };
-  if (isUpdate) value.iscnId = iscnId;
-  const message = {
-    typeUrl: isUpdate ? '/likechain.iscn.MsgUpdateIscnRecord' : '/likechain.iscn.MsgCreateIscnRecord',
-    value,
-  };
-  const fee = await estimateISCNTxGas(message);
-  const txRaw = await client.sign(account.address, [message], fee, '', explicitSignerData);
-  const txBytes = TxRaw.encode(txRaw).finish();
-  const response = await client.broadcastTx(txBytes);
-  assertIsBroadcastTxSuccess(response);
-  return parseISCNTxInfoFromTxSuccess(response);
+async function getSignerData() {
+  const address = await getAddress();
+  const client = await getSigningStargateClient();
+  const { accountNumber, sequence } = await client.getSequence(address);
+  const chainId = await client.getChainId();
+  return { accountNumber, sequence, chainId };
 }
 
 async function getSequence() {
-  const { wallet, account: { address } } = await getWallet();
-  const client = await getSigningClient(wallet);
+  const address = await getAddress();
+  const client = await getSigningStargateClient();
   const { sequence } = await client.getSequence(address);
   return sequence;
 }
 
-async function getSignerData() {
-  const { wallet, account: { address } } = await getWallet();
-  const client = await getSigningClient(wallet);
-  const { accountNumber, sequence } = await client.getSequence(address);
-  const chainId = await client.getChainId();
-  return { accountNumber, sequence, chainId };
+async function getAccountBalance() {
+  const address = await getAddress();
+  const client = await getSigningStargateClient();
+  const balance = await client.getBalance(address, COSMOS_DENOM);
+  return balance;
+}
+
+async function createISCNRecord(payload, signOptions) {
+  const address = await getAddress();
+  const queryClient = await getISCNQueryClient();
+  const signingClient = await getISCNSigningClient();
+  const res = await signingClient.createISCNRecord(address, payload, signOptions);
+  const { transactionHash: txHash } = res;
+  const [iscnId] = await queryClient.queryISCNIdsByTx(txHash);
+  return { txHash, iscnId };
+}
+
+async function estimateISCNFee(data, convertFunc) {
+  const signingClient = await getISCNSigningClient();
+  const gasFee = (await signingClient.estimateISCNTxGas(data)).fee.amount[0].amount;
+  let result = new BigNumber(gasFee);
+  try {
+    const promises = data.map((item) => {
+      const payload = convertFunc(item);
```

Comment:
Just convert `data` outside before calling this function will do? Seems it is a simple `.map()` that this function does not need to know about